app/controllers/AbstractGreeterServiceController.scala

```diff
@@ -0,0 +1,30 @@
+package controllers
+
+import akka.http.scaladsl.model.{HttpRequest, HttpResponse}
+import akka.stream.Materializer
+import io.grpc.examples.helloworld.{GreeterService, GreeterServiceHandler}
+import play.api.inject.Injector
+import play.api.mvc.akkahttp.AkkaHttpHandler
+import play.api.routing.Router
+import play.api.routing.Router.Routes
+
+import scala.concurrent.Future
+
+/**
+ * FIXME: this needs to be generated
+ * - note that it is the play injector we accept - not hardcoding the router to use Guice
+ */
+abstract class AbstractGreeterServiceController(injector: Injector) extends Router with GreeterService {
```

Comment:
I don't think we should call it a Controller. It's a Router and this will confuse the users.

This shows up later on the `routes` file.
```
->     /helloworld.GreeterService   controllers.GreeterServiceController
```

The `->` is for Routers, http verbs for Controller methods.

Reply:
Alright

Reply:
The hard part of naming is that it will be both a router and the service implementation, that it is a router is secondary and isn't really very interesting to the user, it is just how it gets hooked into play.

Can't think of anything better than `AbstractNnnnNnnRouter`, and then user call the implementation whatever they want, we'd probably need some convention to use in samples though.

Reply:
Yeah, that's maybe neater with traits (`GreeterServiceImpl extends GreeterService with GreeterServicePlayRouterSupport` conveys that the fact that it's made available as a Router is secondary).. `abstract class AbstractPlayGreeterService` is getting rather verbose...

Reply:
`with SomethingPlayRouterSupport` sounds a better option

Reply:
You guys are way too late with the bike shedding here! I've already shipped it :shipit: !
https://github.com/akka/akka-grpc/pull/291